packages/contracts-bedrock/src/L2/L1Block.sol

```diff
@@ -177,6 +206,47 @@ contract L1Block is ISemver, IGasToken {
     function setGasPayingToken(address _token, uint8 _decimals, bytes32 _name, bytes32 _symbol) external {
         if (msg.sender != DEPOSITOR_ACCOUNT()) revert NotDepositor();
 
+        _setGasPayingToken(_token, _decimals, _name, _symbol);
+    }
+
+    /// @notice Sets static configuration options for the L2 system. Can only be called by the special
+    ///         depositor account.
+    /// @param _type  The type of configuration to set.
+    /// @param _value The encoded value with which to set the configuration.
+    function setConfig(ConfigType _type, bytes calldata _value) public virtual {
+        if (msg.sender != DEPOSITOR_ACCOUNT()) revert NotDepositor();
+
+        if (_type == ConfigType.SET_GAS_PAYING_TOKEN) {
+            (address token, uint8 decimals, bytes32 name, bytes32 symbol) = StaticConfig.decodeSetGasPayingToken(_value);
+            _setGasPayingToken(token, decimals, name, symbol);
```

Comment:
Move this line inside the `_setGasPayingToken` function as it was in the `L1BlockInterop` contract

Reply:
My reasoning for this was avoid having repeated code between `setConfig` and `setGasPayingToken` also didn't want to encode the params in `setGasPayingToken` just to be decoded in the `_setGasPayingToken` internal function. I'll make the external `setGasPayingToken` to handle the whole process itself and the internal function to use `bytes memory` for the sake of consistency in the `setConfig` then.